src/coreclr/pal/tests/palsuite/threading/SetThreadDescription/test1/test1.cpp

```diff
@@ -0,0 +1,110 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*============================================================
+**
+** Source: test1.cpp
+**
+** Purpose: Test for SetThreadDescription.  Create a thread, call 
+** SetThreadDescription, and then verify that the name of the thread
+** matches what was set.
+**
+**=========================================================*/
+
+#include <palsuite.h>
+
+char * threadName = NULL;
+char * expectedThreadName = NULL;
+char * actualThreadName = new char[256];
+
+DWORD PALAPI SetThreadDescriptionTestThread(LPVOID lpParameter) 
+{
+    HRESULT result;
+    HANDLE palThread = GetCurrentThread();
+    WCHAR wideThreadName[256];
+
+    MultiByteToWideChar(CP_ACP, 0, threadName, strlen(threadName)+1, wideThreadName, 256);
+    SetThreadDescription(palThread, wideThreadName);
+    result = GetThreadDescription(palThread, 256, actualThreadName);
+    return 0;
+}
+
+
+BOOL SetThreadDescriptionTest(char* name, char* expected)
+{
+    BOOL bResult = FALSE;
+    LPSECURITY_ATTRIBUTES lpThreadAttributes = NULL;
+    DWORD dwStackSize = 0; 
+    LPTHREAD_START_ROUTINE lpStartAddress =  &SetThreadDescriptionTestThread;
+    LPVOID lpParameter = (LPVOID)SetThreadDescriptionTestThread;
+    DWORD dwCreationFlags = 0; 
+    DWORD dwThreadId = 0;
+    
+    threadName = name;
+    expectedThreadName = expected;
+
+    HANDLE hThread = CreateThread(lpThreadAttributes, 
+                            dwStackSize, lpStartAddress, lpParameter, 
+                            dwCreationFlags, &dwThreadId );
+   
+    if (hThread != INVALID_HANDLE_VALUE)
+    {
+        WaitForSingleObject(hThread, INFINITE);
+        bResult = strcmp(actualThreadName, expectedThreadName) == 0;
+    }
+    else
+    {
+        Trace("Unable to create SetThreadDescription test thread");
+    }
+
+    return bResult;
+}
+
+BOOL SetThreadDescriptionTests()
+{
+    if (!SetThreadDescriptionTest("Hello, World", "Hello, World"))
+    {
+        Trace("Setting thread name failed");
+        return false;
+    }
+    
+    // verify that thread name truncations works correctly on linux on macos
+    char * threadName;
+    char * expected;
+    #if defined(__linux__)
+    threadName = "linuxstring15characters";
+    expected = "linuxstring15ch";
+    #endif
+
+    #if defined(__APPLE__)
+    threadName = "appplestring63charactersappplestring63charactersappplestring63characters";
+    expected = "appplestring63charactersappplestring63charactersappplestring63c";
+    #endif
```

Comment:
nit: `threadName` could be common with same control flow as above:

```c
threadName = "aaaaaa_15chars_aaaaaaa_31chars_aaaaaaaaaaaaaaaaaaaaa_61chars_aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";

#ifdef __APPLE__
expected = "aaaaaa_15chars_aaaaaaa_31chars_aaaaaaaaaaaaaaaaaaaaa_61chars";
#else
expected = "aaaaaa_15chars";
#endif
```
(it would make it a bit easier to handle other platforms like SunOS, which has 31 characters limit)